### PR TITLE
feat(nm): add firewall NAT rules update on configuration change

### PR DESCRIPTION
**Pull request by marcellorinaldo**
This PR ensures that all the interfaces in NAT will have proper firewall rules configured for the existing WAN interfaces.

**Related Issue:** N/A.

**Description of the solution adopted:** N/A.

**Screenshots:** N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by marcellorinaldo** [comment]
Currently testing.

**Comment by marcellorinaldo** [comment]
Testing made on Raspberry PI 4 - *generic-arm32* profile - `dnsmasq` as DHCP server provider.

| `wlan0` status | Address | DNS passed | Ping 8.8.8.8 | Ping google |
| - | - | - | - | - |
|dhcp, nat, dns|yes|yes|yes|yes|
|dhcp, nat|yes|yes (wrong)|yes|no|
|dhcp, dns|yes|yes|no|no|
|dhcp|yes|yes (wrong)|no|no|
|nat|manual|no|yes|no (yes if setting DNS manually)|

- **Address**: if the client is assigned a correct IP address from DHCP server.
- **DNS passed**: if the DNS server was passed to client, controlled inspecting file `/etc/resolv.conf` containing the entry `nameserver 172.16.1.1`, where 172.16.1.1 is the address of the Raspberry PI 4.

`dnsmasq` is currently configured to not allow to disable DNS server forwarding to DHCP clients: if the property `dhcp-server-option=6,` is not set then default of `dhp-server-option=6,0.0.0.0` is used which means "forward DNS found on localhost".

This issue will be solved here https://github.com/eclipse/kura/pull/4515.

**Comment by marcellorinaldo** [comment]
After rebase, you should be able to test also the pass DNS feature. Only thing missing is https://github.com/eclipse/kura/pull/4529.

File: kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java [hunk 1]
```diff
@@ -314,6 +318,59 @@ private void writeNetworkConfigurationSettings(Map<String, Object> networkProper
...
+                wanInterfaces.add(interfaceName);
+            }
+        });
+
+        interfaceNames.forEach(interfaceName -> {
+            if (isNatValid(interfaceName)) {
+                natInterfaces.add(interfaceName);
+            }
+        });
+
```
Comment (pierantoniomerlino) [inline kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java:334]: Does it make sense to merge these loops?

**Review by pierantoniomerlino — APPROVED** [review]

**Comment by pierantoniomerlino** [comment]
The code is ok. I tested it with a RPi and multiple WAN/LAN interfaces.

File: kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java (hunk 1 above)
Comment (marcellorinaldo) [reply kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/configuration/NMConfigurationServiceImpl.java:334]: Totally, will make this change soon

**Pull request merged**